src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,31 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+	
+	private String value;
+    private boolean isPrivate;
+	
+	protected Contact(){
+	}
+	public String toString(){
+		return value;
+	}
+	
+	@Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
```

Comment:
can you see why this is a problem? you can now potentially compare `Address` with `Email`, why?